core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/FastDoubleParserTests.kt

```diff
@@ -0,0 +1,163 @@
+package org.jetbrains.kotlinx.dataframe.io
+
+import io.kotest.matchers.collections.shouldContainInOrder
+import org.jetbrains.kotlinx.dataframe.api.ParserOptions
+import org.jetbrains.kotlinx.dataframe.impl.io.FastDoubleParser
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.Locale
+
+private const val LOG_LEVEL = "org.slf4j.simpleLogger.defaultLogLevel"
+
+class FastDoubleParserTests {
+
+    private var loggerBefore: String? = null
+
+    @Before
+    fun setLogger() {
+        loggerBefore = System.getProperty(LOG_LEVEL)
+        System.setProperty(LOG_LEVEL, "debug")
+    }
+
+    @After
+    fun restoreLogger() {
+        if (loggerBefore != null) {
+            System.setProperty(LOG_LEVEL, loggerBefore)
+        }
+    }
+
+    @Test
+    fun `can fast parse doubles`() {
+        val parser = FastDoubleParser(ParserOptions(locale = Locale.ROOT, useFastDoubleParser = true))
+
+        val numbers = listOf(
+            "+12.45",
+            "-13.35",
+            "100123.35",
+            "-204,235.23",
+            "1.234e3",
+            "3e-04", // failed with old double parser
+            "nAn",
+            "-N/a",
+            "inf",
+            "-InfinIty",
+        )
+
+        val expectedDoubles = listOf(
+            12.45,
+            -13.35,
+            100_123.35,
+            -204_235.23,
+            1.234e3,
+            3e-04,
+            Double.NaN,
+            -Double.NaN,
+            Double.POSITIVE_INFINITY,
+            Double.NEGATIVE_INFINITY,
+        )
+
+        // CharSequence
+        numbers.map { parser.parseOrNull(it) }.shouldContainInOrder(expectedDoubles)
+
+        // CharArray
+        numbers.map { parser.parseOrNull(it.toCharArray()) }.shouldContainInOrder(expectedDoubles)
+
+        // ByteArray
+        numbers.map { parser.parseOrNull(it.toByteArray()) }.shouldContainInOrder(expectedDoubles)
+    }
+
+    @Test
+    fun `can fast parse german locale`() {
+        val parser = FastDoubleParser(ParserOptions(locale = Locale.GERMANY, useFastDoubleParser = true))
+
+        val numbers = listOf(
+            "12,45",
+            "-13,35",
+            "100.123,35",
+            "-204.235,23",
+            "1,234e3",
+        )
+
+        val expectedDoubles = listOf(
+            12.45,
+            -13.35,
+            100_123.35,
+            -204_235.23,
+            1.234e3,
+        )
+
+        // CharSequence
+        numbers.map { parser.parseOrNull(it) }.shouldContainInOrder(expectedDoubles)
+
+        // CharArray
+        numbers.map { parser.parseOrNull(it.toCharArray()) }.shouldContainInOrder(expectedDoubles)
+
+        // ByteArray
+        numbers.map { parser.parseOrNull(it.toByteArray()) }.shouldContainInOrder(expectedDoubles)
+    }
+
+    @Test
+    fun `can fast parse french locale`() {
+        val parser = FastDoubleParser(ParserOptions(locale = Locale.FRANCE, useFastDoubleParser = true))
```

Comment:
I started to think: could we write a test, which just generates checks for all possible locales and constants are obtained from Double values -> String -> parseDouble again?

Reply:
"all" locales? XD I can try, but that's like hundreds of locales and may be system-dependent.

Reply:
Also, there are some locales like Arabic where NumberFormat itself fails to add "-" in front of negative numbers :/ It's fixed in later Java versions, but not 8 yet.

Reply:
Okay this was actually a really good idea! It made me catch two separate bugs which I reported to [FastDoubleParser](https://github.com/wrandelshofer/FastDoubleParser/issues/82).
- Minus signs in RTL languages, like Arabic, are (in Java 8) marked with the [Arabic letter mark](https://en.wikipedia.org/wiki/Arabic_letter_mark) instead of, you know, a minus sign... FastDoubleParser cannot handle this, because the character is a format symbol, not a normal one. `NumberFormat` does handle it, so to align our behavior with theirs, I now manually check for format symbol minus signs and replace them with '-' before parsing.
- In Finnish and other langauges, they use `exponentSeparator='e'` and `nan="epäluku"`, which breaks as both contain the letter 'e'. I've reported this and luckily our fallback method catches it.